numerical/utils/linalg.py

```diff
@@ -22,3 +22,13 @@ def multi_dot2(*vectors, flatten=False, reshape=False):
     if reshape:
         md = md.reshape(-1, 1)
     return md
+
+
+def polar2cartesian(x):
```

Comment:
Consider to use https://docs.python.org/3/library/cmath.html#conversions-to-and-from-polar-coordinates instead

Reply:
cmath library in python does not support vector computation, so I can not evaluate the vector of variables without loop. I advise you to forgot cmath library in python for the cases of big computations.